core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/HealthMonitor.scala

```diff
@@ -132,11 +138,21 @@ class HealthMonitor private (val slickDataSource: SlickDataSource, val googleSer
 
   /**
     * Checks Cromwell status.
-    * TODO: Cromwell doesn't have a status API yet so this is stubbed out for now.
+    * Calls each Cromwell's /status endpoint and returns true if all Cromwells return true
     */
   private def checkCromwell: Future[SubsystemStatus] = {
-    logger.debug("Checking Cromwell...")
-    Future.successful(OkStatus)
+    logger.debug("Checking Cromwell(s)...")
+
+    def annotateSubsystem(serviceId: ExecutionServiceId, execSubsystem: String, subsystemStatus: SubsystemStatus): SubsystemStatus = {
+      subsystemStatus.copy(messages = subsystemStatus.messages.map { msgList => msgList.map { msg => s"${serviceId.id}-$execSubsystem: $msg" } })
+    }
+
+    // Note: calls to `foldMap` depend on SubsystemStatusMonoid
```

Comment:
Nice way to fold a `List[Future[List[SubsystemStatus]]]` to a `Future[SubsystemStatus]`

Reply:
my mother was always bad at folding maps